Rserve/src/Rserv.c

```diff
@@ -1123,7 +1123,8 @@ static SEXP decode_to_SEXP(unsigned int **buf, int *UPC)
 			(*UPC)++;
 			i = 0;
 			while (i < vl) {
-				LOGICAL(val)[i] = cb[i];
+				//      LOGICAL(val)[i] = cb[i];
+                LOGICAL(val)[i] = (cb[i] == 1) ? TRUE : ((cb[i] == 0) ? FALSE : NA_LOGICAL);
```

Comment:
When a REXPLogical vector is passed to Rserve using Buffer, TRUE = 1, FALSE = 0, NA = -124
cb is buffer which would contain the values passed from Rserve client
while parsing this buffer and creating a Logical vector for R, we check for correct values and assing them in in LOGICAL vector

This is the Fix I found in latest version of Rserve 1.7 and I have used the same logic for parsing the Logical vector